### PR TITLE
HTTP tx submission - make bulk and JSON modes consistent

### PR DESCRIPTION
**Pull request by wkaluza**

### CODE REVIEW

File: libs/ledger/src/chaincode/contract_http_interface.cpp [hunk 1]
```diff
@@ -332,22 +365,14 @@ ContractHttpInterface::SubmitTxStatus ContractHttpInterface::SubmitJsonTx(
   FETCH_LOG_DEBUG(LOGGING_NAME, "NEW TRANSACTION RECEIVED");
   FETCH_LOG_DEBUG(LOGGING_NAME, request.body());
 
-  FETCH_UNUSED(expected_contract);
```
Comment (ejfitzgerald) [inline libs/ledger/src/chaincode/contract_http_interface.cpp:335]: Let's have a chat about this F2F. However, the original intension here (and it appears to have been lost in `master`) is that based on the submission endpoint, there is a restriction on the target contract that can be submitted.

The principal idea here is that by imposing this restriction we can more closely monitor the transactions that are being submitted externally.

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]
Looks good, lets catch up regarding the `expected_contract` stuff

File: libs/ledger/src/chaincode/contract_http_interface.cpp (hunk 1 above)
Comment (wkaluza) [reply libs/ledger/src/chaincode/contract_http_interface.cpp:335]: Ok, good to know. What caught my eye initially was the differences between how we treat 'bulk' and JSON submissions, so I would like to have a word about that too.

**Review by ejfitzgerald — APPROVED** [review]
As discussed - we will submit this change for the short term and then revisit adding back the restrictions into the HTTP endpoint

**Pull request merged**